src/record/record.js

```diff
@@ -151,24 +151,37 @@ Record.prototype.set = function( pathOrData, dataOrCallback, callback ) {
 	var newValue = jsonPath.set( oldValue, path, data, this._options.recordDeepCopy );
 
 	if ( oldValue === newValue ) {
+		if ( typeof callback === 'function' ) {
+			if (isConnected( this._client )) {
+				utils.setImmediate( callback, null );
+			} else {
+				utils.setImmediate( callback, 'Connection error: error updating record as connection was closed' );
+			}
+		}
 		return this;
 	}
 
 	var config;
-	if( callback !== undefined ) {
+	if( typeof callback === 'function' ) {
 		config = {};
 		config.writeSuccess = true;
 		this._setUpCallback(this.version, callback)
-		var connectionState = this._client.getConnectionState();
-		if( connectionState === C.CONNECTION_STATE.CLOSED || connectionState === C.CONNECTION_STATE.RECONNECTING ) {
-			callback( 'Connection error: error updating record as connection was closed' );
+
+		if (!isConnected( this._client )) {
+			utils.setImmediate( callback, 'Connection error: error updating record as connection was closed' );
 		}
 	}
+
 	this._sendUpdate( path, data, config );
 	this._applyChange( newValue );
 	return this;
 };
 
+function isConnected( client ) {
+	var connectionState = client.getConnectionState();
```

Comment:
No I meant if you do a set before login occurs, state is authenticating or awaiting authentication